### DISCUSSION TITLE
setting which visualizers are open from reading text file?

### DISCUSSION THREAD

**Discussion by edmundchong** [bonsai-rx/bonsai#1934] [Q&A] [answered]
Hello! I was wondering if there exists any functionality that sets the display of visualizers (which nodes have active visualizers, and positions / sizes of these visualizers etc.) based on reading some inputs e.g. from a text file?

**Comment by bruno-f-cruz** [comment]
Not at the moment (well I guess one could toggle between streams of the same time with some cleverness :P). But this is kinda the flavor of what I was thinking about here:

https://github.com/bonsai-rx/bonsai/issues/1862

**Reply by edmundchong** [reply]
@bruno-f-cruz yes something like that--for my use case it's that different subjects in the experiment can have slightly different data plots that I'm interested in showing in real time. Was also wondering--would there not be a way to modify the .layout XML file (which I presume controls visualizer layout?)

**Reply by bruno-f-cruz** [reply]
You can, but its rather error prone. Why not just have different layout files and toggle between them?

**Reply by edmundchong** [reply]
That would be great--for now the layout file that is chosen by default is just the one with the identical name to the bonsai script being executed, correct? How do I set that dynamically?

**Reply by bruno-f-cruz** [reply, marked as answer]
Via the command line interface
https://bonsai-rx.org/docs/articles/cli.html#specifying-layout-files-in-application-mode

**Reply by edmundchong** [reply]
very useful thanks! this will work great for my current use case but i second your suggestion #1862 for ultimate flexibility :)
